lib/src/main/java/io/ably/lib/realtime/Channel.java

```diff
@@ -475,6 +477,24 @@ private void failQueuedMessages(ErrorInfo reason) {
 		queuedMessages.clear();
 	}
 
+	private void replacePlaceholderParams(Param[] params) throws AblyException {
+		Param param;
+		for (int i = 0; i < params.length; i++) {
+			param = params[i];
+
+			if (KEY_UNTIL_ATTACH.equals(param.key) && "true".equals(param.value)) {
+				if (state != ChannelState.attached) {
+					throw AblyException.fromErrorInfo(new ErrorInfo("option untilAttach requires the channel to be attached", 40000, 400));
+				}
+
+				params[i] = new Param(KEY_FROM_SERIAL, attachSerial);
+			}
+		}
+	}
+
+
+	private static final String KEY_UNTIL_ATTACH = "untilAttach";
+	private static final String KEY_FROM_SERIAL = "from_serial";
```

Comment:
FYI @gokhanbarisaker we changed all query string params from under_score case to camelCase a little while ago but did not roll it out everywhere, hence this comment...